From 14ad91658fd296e34bb9e833281e72c871bfb189 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex@linutronix.de>
Date: Tue, 3 May 2022 12:31:50 +0200
Subject: [PATCH] video: restore ability to disable fb accel via hint

Somewhere in code refactoring between .20 and .22 this check
was lost, and so the hint had no effect anymore.

Upstream-Status: Submitted [https://github.com/libsdl-org/SDL/pull/5611]
Signed-off-by: Alexander Kanavin <alex@linutronix.de>
---
 src/video/SDL_video.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/video/SDL_video.c b/src/video/SDL_video.c
index 93c803e..cbe7616 100644
--- a/src/video/SDL_video.c
+++ b/src/video/SDL_video.c
@@ -2503,6 +2503,14 @@ SDL_CreateWindowFramebuffer(SDL_Window * window)
     if (!_this->checked_texture_framebuffer) {
         SDL_bool attempt_texture_framebuffer = SDL_TRUE;
 
+        /* See if the user or application wants to specifically disable the framebuffer */
+        const char *hint = SDL_GetHint(SDL_HINT_FRAMEBUFFER_ACCELERATION);
+        if (hint) {
+            if (*hint == '0' || SDL_strcasecmp(hint, "false") == 0) {
+                attempt_texture_framebuffer = SDL_FALSE;
+            }
+        }
+
         if (_this->is_dummy) {  /* dummy driver never has GPU support, of course. */
             attempt_texture_framebuffer = SDL_FALSE;
         }
-- 
2.30.2

